modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java

```diff
@@ -219,26 +243,26 @@ private static int getLogFileSequenceNumber() {
     @Override
     protected List<Attribute> init(AbstractDefinition inputDefinition, ExpressionExecutor[]
             attributeExpressionExecutors, ConfigReader configReader, SiddhiAppContext siddhiAppContext) {
-        executorService = siddhiAppContext.getExecutorService();
 
+        executorService = siddhiAppContext.getExecutorService();
         siddhiAppContextName = siddhiAppContext.getName();
 
-        if (attributeExpressionLength == 2) {
+        if (attributeExpressionExecutors.length >= 2 && attributeExpressionExecutors.length <= 4) {
+
             if (!(attributeExpressionExecutors[0] instanceof VariableExpressionExecutor)) {
                 throw new SiddhiAppValidationException("iijTimeStamp has to be a variable but found " +
                         this.attributeExpressionExecutors[0].getClass().getCanonicalName());
             }
 
-            if (attributeExpressionExecutors[0].getReturnType() == Attribute.Type.LONG) {
-
-            } else {
+            if (!(attributeExpressionExecutors[0].getReturnType() == Attribute.Type.LONG)) {
                 throw new SiddhiAppValidationException("iijTimestamp is expected to be long but found "
                         + attributeExpressionExecutors[0].getReturnType());
             }
 
             if (!(attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor)) {
-                throw new SiddhiAppValidationException("second parameter has to be constant but " +
-                        "found " + this.attributeExpressionExecutors[1].getClass().getCanonicalName());
+                throw new SiddhiAppValidationException("Type has to be constant(\"throughput\"or " +
+                        "\"latency\" or \"both\") but " + "found " +
```

Comment:
You could change this to

"\"latency\" or \"both\") but  found " +

Reply:
Fixed https://github.com/wso2/product-sp/pull/880/commits/bcb8b40e0439f8be16498c95eb514241f9e723df